### ISSUE TITLE
Trait bound not satisfied

### ISSUE THREAD

**Issue by jneumann** [pierreyoda/solar-rust#2]
When I try to build this with rust version 1.9.0, I get the error in the attached screen shot.
<img width="1361" alt="screen shot 2016-07-01 at 10 03 22 pm" src="https://cloud.githubusercontent.com/assets/4145595/16537905/a63ff584-3fd7-11e6-9f25-3477befd709f.png">

**Comment by pierreyoda** [comment]
Yes I've been having a lot of trouble and frustration with interfacing traits between piston's various internal crates and I thought I had resolved the problem but it may be so only on my computer (using rustc stable 1.9.0 on Windows 64 bits with mingw), I'll look into it with a fresh build ASAP ! Thank you for your interest.
